packages/core/admin/admin/src/content-manager/components/DynamicZone/components/Component/utils/useMainValue.js

```diff
@@ -0,0 +1,24 @@
+import { useMemo } from 'react';
+import { get, toString } from 'lodash';
+import { useCMEditViewDataManager } from '@strapi/helper-plugin';
+
+export function getDisplayedValue(modifiedData, componentFieldPath, mainField) {
+  let displayedValue = toString(get(modifiedData, [...componentFieldPath, mainField], ''));
+
+  if (displayedValue.length > 50) {
+    displayedValue = `${displayedValue.substring(0, 50)}...`;
+  }
```

Comment:
As mentioned in my previous comment, I think it would be best to wrap the entire content similar to the `AccordionToggle` and this should not be needed.

Reply:
Ok, I removed the limit checking and the crop.